src/__tests__/ProductListItem.spec.js

```diff
@@ -0,0 +1,18 @@
+import { render, screen, fireEvent } from "@testing-library/vue";
+import ProductListItem from "../components/product/ProductListItem.vue";
+
+describe("ProductListItem.vue", () => {
+  it("Edit button is visible", () => {
+    render(ProductListItem);
+    expect(screen.getByRole("edit")).toBeVisible();
+  });
+
+  it("Edit button clicked and emmit has been passed", async () => {
+    const { emitted } = await render(ProductListItem);
+    const button = screen.getByRole("edit");
+    fireEvent.click(button);
```

Comment:
Event firing is asynchronous, hence, every call to `fireEvent` should be `await`ed. 😉